lasagne/layers/corrmm.py

```diff
@@ -149,25 +148,21 @@ def __init__(self, incoming, num_filters, filter_size, stride=(1, 1),
         self.untie_biases = untie_biases
         self.flip_filters = flip_filters
 
-        if border_mode is not None and pad is not None:
-            raise RuntimeError("You cannot specify both 'border_mode' and "
-                               "'pad'. To avoid ambiguity, please specify "
-                               "only one of them.")
-        elif border_mode is None and pad is None:
+        if pad is None:
             # no option specified, default to valid mode
             self.pad = (0, 0)
-        elif border_mode is not None:
-            if border_mode == 'valid':
+        elif isinstance(pad, basestring):
```

Comment:
That's not really needed. Also you probably shouldn't replace `full` with a custom amount of padding, but leave it up to [GpuCorrMM](http://deeplearning.net/software/theano/library/sandbox/cuda/op.html#theano.sandbox.cuda.blas.GpuCorrMM) (but that was wrong before). I'd suggest:
```python
if pad == 'valid':
    self.pad = (0, 0)
elif pad == 'full':
    self.pad = 'full'
elif pad == 'same':
    self.pad = 'half'
else:
    self.pad = as_tuple(pad, 2)
```